izanami-frontend/src/pages/users.tsx

```diff
@@ -83,24 +83,12 @@ export function Users() {
       },
     }
   );
-  const inviteUserMutation = useMutation<
-    { invitationUrl?: string } | null,
-    Error,
-    { admin: boolean; email: string; rights: TRights; userToCopy: string }
-  >((data) => {
-    if (data.userToCopy) {
-      return queryUser(data.userToCopy)
-        .then((res) => {
-          data.rights = res.rights;
-          return createInvitation(data.email, data.admin, data.rights);
-        })
-        .catch((error) => {
-          throw new Error(error);
-        });
-    } else {
-      return createInvitation(data.email, data.admin, data.rights);
-    }
-  });
+  const inviteUserMutation = useMutation(
+    (data: { email: string; admin: boolean; rights: TRights }) =>
+      createInvitation(data.email, data.admin, data.rights)
+  );
+
+  const readUser = useMutation((user: string) => queryUser(user));
```

Comment:
Since queryUser doesn't change anything server side it should be a query. However I would prefer not having a query at all, and chaining it with `createInvitation` as it was done before (sorry, I told you that userToCopy wasn't necessary anymore, however you'll need it to know it you need to queryUser or not).

Having this logic in mutation makes more sense to me, rather than bloating jsx with this logic.